cypress/e2e/tests-ejercicio/login.cy.js

```diff
@@ -1,21 +1,91 @@
 const LoginPage = require("../../pages/saucedemo/login");
 
 describe("Login", () => {
-    it("Check Login", () => {
+    const usernames = [
+        "standard_user",
+        "locked_out_user",
+        "problem_user",
+        "performance_glitch_user",
+        "error_user",
+        "visual_user"
+    ];
+
+    beforeEach(() => {
         cy.visit("/")
+    })
+    it("Check Login", () => {
         cy.url().should("eq", "https://www.saucedemo.com/")
         cy.title().should("eq", "Swag Labs")
     })
 
     it("Login with valid credentials", () => {
-        cy.login("standard_user", "secret_sauce")
-        cy.url().should("contain",  "/inventory.html")
+        // cy.login("standard_user", "secret_sauce")
+        LoginPage.login("standard_user", "secret_sauce")
+        cy.url().should("contain", "/inventory.html")
         cy.title().should("eq", "Swag Labs")
     })
 
     it("Login with invalid credentials", () => {
-        cy.login("invalid", "invalid")
-        cy.get('[data-test="error"]')
+        LoginPage.login("Invalid", "Invalid")
+        LoginPage.createErrorMessage().should("exist")
+        LoginPage.createErrorMessage().should("have.text", "Epic sadface: Username and password do not match any user in this service")
+        cy.url().should("eq", "https://www.saucedemo.com/")
+    })
+
+    it('Login without credentials', () => {
+        LoginPage.clickLogin()
+        LoginPage.createErrorMessage().should("exist")
+        LoginPage.createErrorMessage().should("have.text", "Epic sadface: Username is required")
+        cy.url().should("eq", "https://www.saucedemo.com/")
+    });
+
+    it('Login without username', () => {
+        LoginPage.typePassword("Invalid")
+        LoginPage.clickLogin()
+        LoginPage.createErrorMessage().should("exist")
+        LoginPage.createErrorMessage().should("have.text", "Epic sadface: Username is required")
+        cy.url().should("eq", "https://www.saucedemo.com/")
+    });
+
+    it('Login without password', () => {
+        LoginPage.typeUsername("Invalid")
+        LoginPage.clickLogin()
+        LoginPage.createErrorMessage().should("exist")
+        LoginPage.createErrorMessage().should("have.text", "Epic sadface: Password is required")
         cy.url().should("eq", "https://www.saucedemo.com/")
+    });
+
+    // Test every credential
+    usernames.forEach((username) => {
```

Comment:
Me gustó mucho esta implementación! Solo te quiero dejar unas observaciones:
- En vez de usar un if y else, seria mejor dejar definido la prueba de locked_out_user en un it fuera de este set de pruebas
- En vez de usar el nombre 'Login with all the credentials user' hubiera puesto simplemente: 'Login with credential user:' (anidando el nombre de la credencial al final tal cual lo hiciste), pero con esto dejamos en claro que en cada test no estamos probando todas las credenciales, es el conjunto de pruebas el que lo hace.
De todas formas me gusta lo que está hecho! el primer comentario si me parece que habria que cumplirlo pero en cuanto al segundo es solo una cuestión de estética. Ojo también con el test repetido de standard_user, estarias ejecutando la misma prueba dos veces indirectamente :+1: 

Buen trabajo!

Reply:
Okey, lo corrijo lo que me has comentado en el próximo PR